cmd/aserto-scim/main.go

```diff
@@ -32,12 +33,30 @@ var cmdRun = &cobra.Command{
 	Use:   "run [args]",
 	Short: "Start SCIM service",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return app.Run(flagConfigPath, os.Stdout, os.Stderr)
+		srv, err := app.NewSCIMServer(flagConfigPath, os.Stdout, os.Stderr)
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			if err := srv.Run(); err != nil {
+				log.Printf("Error running SCIM server: %v", err)
+			}
+		}()
+
+		stop := make(chan os.Signal, 1)
```

Comment:
It's better to use an error group to start the go routine and wait for it.
You can construct the group with a context that responds to signals (see [here](https://github.com/aserto-dev/directory/blob/70143346b33880dab8548dd0b632415fb478d864/pkg/cc/context.go#L18).
Then use the group to run the go routine and then wait on the context `Done()` channel.
When that returns, call `Shutdown` and then `Wait()` on the error group.